pyfr/readers/native.py

```diff
@@ -166,6 +176,29 @@ def __len__(self):
         return len(self._npf.files)
 
 
+class PyFRH5Reader(PyFRBaseReader):
+    """Does everything that file reader does but with a h5py file"""
+    def __init__(self, fname):
+        self._file = h5py.File(fname, 'r')
+
+    def __getitem__(self, aname):
+        ret = self._file[aname]
+
+        if self._file[aname].shape == ():
```

Comment:
Probably want to use `ret` here.  While h5py is probably not stupid enough to read the entire file it is certainly going to be more work than using `ret`.